Adds basic Toolbar component tests.

BE-Webdesign opened the pull request:
Adds basic Toolbar component tests. Related to progress on #641.

**Testing Instructions**
Run npm i && npm run test-unit ensure tests pass. Change component logic
to ensure tests fail as they should.

BE-Webdesign committed:
Adds basic Toolbar component tests.

Adds basic Toolbar component tests. Related to progress on #641.

**Testing Instructions**
Run npm i && npm run test-unit ensure tests pass. Change component logic
to ensure tests fail as they should.

aduth commented on components/toolbar/test/index.js:
```diff
@@ -0,0 +1,83 @@
+			const toolbar = shallow( <Toolbar controls={ controls } /> );
+			const listItem = toolbar.find( 'IconButton' );
+			expect( toolbar.type() ).to.equal( 'ul' );
+			expect( listItem.props().icon ).to.equal( 'wordpress' );
```
Minor: `listItem.prop( 'icon' )` more explicit. Same goes for most of the following lines. Else we could test:

```js
expect( listItem.props() ).to.eql( {
	// ...keys
} );
```

BE-Webdesign replied:
Yeah, I thought I updated these, guess not lol. Will change.  Should it be changed to the one command style?

```
expect( listItem.props() ).to.eql( {
	// ...keys
} );
```

aduth replied:
I don't have a preference.

aduth commented on components/toolbar/test/index.js:
```diff
@@ -0,0 +1,83 @@
+			const toolbar = shallow( <Toolbar controls={ controls } /> );
+			const listItem = toolbar.find( 'IconButton' );
+			listItem.simulate( 'click', { stopPropagation: () => undefined } );
+			expect( clicked ).to.be.true();
```
Sinon spies are great for this sort of test:

http://sinonjs.org/releases/v2.3.2/spies/

Appears we're now using `sinon-test` as of #924, so it'd be matter of:

```js
it( 'should call the clickHandler on click.', sinon.test( function() {
	const clickHandler = this.spy();
	// ...
	expect( clickHandler ).to.have.been.calledOnce;
} );
```

See also: https://github.com/domenic/sinon-chai

BE-Webdesign replied:
Yeah, I realized sinon was being used in our effects tests, so I will use that here as well in the revisions.

aduth reviewed (approved):
Thanks for working on these tests 👍

BE-Webdesign committed:
Cleaning up tests.

Cleans up the tests to make use of spies.  This also changes the
`props()` convention to check whether the keys are included in the
object.  include is used instead of equal to ensure that the tests do
not look at all props like children. Adds in a correction to the function
signature of Toolbar so that the component renders properly.

BE-Webdesign commented:
@iseulde If you could take a peek at this to make sure this aligns with the changes you made, it would be appreciated.
